nap/utils.py

```diff
@@ -60,19 +51,21 @@ def flatten_params(k, vs):
 
         # since we can have more than one value for a single key, we use a
         # tuple of two tuples instead of a dictionary
-        params_tuple = tuple(itertools.chain(*flat_params))
-        param_string = urllib.parse.urlencode(params_tuple)
+        params_tuple = tuple(sorted(itertools.chain(*flat_params), key=lambda key_value: key_value[0]))
```

Comment:
I'd argue that we should always preserve the order of all params since that could have significance as well (priority, defaulting, etc.) but we're already abandoning that in favor of being more likely to get a cache hit. It sounds like we're somewhat arbitrarily drawing a line about where we want to re-order things.